### PR TITLE
docs: make the comments on using the new confirmTransaction clearer

### PR DESCRIPTION
**Pull request by GHesericsu**
#### Problem
Connection.confirmTransaction using TransactionSignature is deprecated but the docs comment on using the new confirmTransaction wasn't clear

#### Summary of Changes
Clearer comment on using the new confirmTransaction method

Fixes #

### CODE REVIEW

File: web3.js/src/connection.ts [hunk 1]
```diff
@@ -3603,7 +3603,7 @@ export class Connection {
     commitment?: Commitment,
   ): Promise<RpcResponseAndContext<SignatureResult>>;
 
-  /** @deprecated Instead, call `confirmTransaction` using a `TransactionConfirmationConfig` */
+  /** @deprecated Instead, call `confirmTransaction` and pass in `BlockheightBasedTransactionConfirmationStrategy` or `DurableNonceTransactionConfirmationStrategy`  */
```
Comment (steveluscher) [inline web3.js/src/connection.ts:3606]: So, one thing that this introduces is the possibility that – when we add a third strategy – someone could forget to update this comment.

Does this work?

1. Extract `BlockheightBasedTransactionConfirmationStrategy | DurableNonceTransactionConfirmationStrategy` into a union type called `TransactionConfirmationStrategy`.
2. Refer to it here using `{@link TransactionConfirmationStrategy}`

I _think_ that should make the docs autolink to that type, where folks can see the up-to-date list of strategies.

**Review by steveluscher — CHANGES_REQUESTED** [review]

**Comment by steveluscher** [comment]
Thanks for the submission!

File: web3.js/src/connection.ts (hunk 1 above)
Comment (GHesericsu) [reply web3.js/src/connection.ts]: Hi @steveluscher thanks for the suggestion. I updated the PR. Hope this works. My comment describing `TransactionConfirmationStrategy` may need work. Thanks.

File: web3.js/src/connection.ts [hunk 2]
```diff
@@ -3603,13 +3610,11 @@ export class Connection {
...
   confirmTransaction(
-    strategy:
-      | BlockheightBasedTransactionConfirmationStrategy
-      | DurableNonceTransactionConfirmationStrategy,
+    strategy: TransactionConfirmationStrategy,
     commitment?: Commitment,
   ): Promise<RpcResponseAndContext<SignatureResult>>;
 
-  /** @deprecated Instead, call `confirmTransaction` using a `TransactionConfirmationConfig` */
+  /** @deprecated Instead, call `confirmTransaction` and pass in {@link TransactionConfirmationStrategy}  @ */
```
Comment (steveluscher) [inline web3.js/src/connection.ts:3617]: What does the trailing `@` do here? Was that an accident?

File: web3.js/src/utils/send-and-confirm-raw-transaction.ts [hunk 3]
```diff
@@ -15,14 +16,14 @@ import type {ConfirmOptions} from '../connection';
...
- * @param {BlockheightBasedTransactionConfirmationStrategy} confirmationStrategy
+ * @param {TransactionConfirmationStrategy} confirmationStrategy
  * @param {ConfirmOptions} [options]
  * @returns {Promise<TransactionSignature>}
  */
 export async function sendAndConfirmRawTransaction(
   connection: Connection,
   rawTransaction: Buffer,
-  confirmationStrategy: BlockheightBasedTransactionConfirmationStrategy,
+  confirmationStrategy: TransactionConfirmationStrategy,
```
Comment (steveluscher) [inline web3.js/src/utils/send-and-confirm-raw-transaction.ts:26]: Ooh, nice. You found a bug!

File: web3.js/src/connection.ts (hunk 2 above)
Comment (GHesericsu) [reply web3.js/src/connection.ts]: ah typo. Fixed.

**Pull request merged**